packages/xsnap/src/avaAssertXS.js

```diff
@@ -0,0 +1,315 @@
+/* eslint-disable no-await-in-loop */
+// @ts-check
+
+/** global print */
+
+const { freeze, keys } = Object;
+
+// ack Paul Roub Aug 2014
+// https://stackoverflow.com/a/25456134/7963
+/** @type {(x: unknown, y: unknown) => boolean } */
+function deepEqual(x, y) {
+  if (x === y) {
```

Comment:
I looked into a dependency on [concordance](https://github.com/concordancejs/concordance#readme), but I don't have rollup / bundleSource in place in the relevant part of the code, so it would be a bit of a hassle.

I could perhaps add it but I'm hesitant to depend on 6Mb (54Kloc) of code for deepEqual.

I'd like to think that a conservative approximation (one that is at least as discriminating as ava's) would be close enough for testing purposes. The current code fails that in at least the following ways:
  - concordance is sensitive to property order and
  - concordance handles recursive structures to some extent

I suppose that means `t.notDeepEqual` is _less_ discriminating - for example, it would consider 2 `Map`s distinct even though they have the same keys and values (in order).

```
info Direct dependencies
└─ concordance@5.0.2
info All dependencies
├─ blueimp-md5@2.18.0
├─ concordance@5.0.2
├─ date-time@3.1.0
├─ esutils@2.0.3
├─ fast-diff@1.2.0
├─ js-string-escape@1.0.1
├─ lodash@4.17.21
├─ lru-cache@6.0.0
├─ md5-hex@3.0.1
├─ semver@7.3.4
├─ time-zone@1.0.0
├─ well-known-symbols@2.0.0
└─ yallist@4.0.0
```

Reply:
for follow-up, see #2507